src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,184 @@
+            let dailyData: { [key: string]: DailyDataQueryResult } = {};
+            results.forEach((result, index) => {
+                dailyData[props.dataTypes[index].dailyDataType] = result;
+            });
+
+            let computedResults: RelativeActivityResult[] = [];
+            let dayKey = getDayKey(date!);
+            props.dataTypes.forEach(dataType => {
+                let data = dailyData![dataType.dailyDataType];
+                if (!data || !data[dayKey]) {
+                    return;
+                }
+                if (props.specifyThresholds && !dataType.threshold) {
+                    return;
+                }
+
+                let threshold = dataType.threshold;
+                if (!props.specifyThresholds) {
+                    let sumValues = 0;
+                    let totalValues = 0;
+                    for (var i = 1; i < 31; i++) {
+                        let key = getDayKey(add(date!, { days: -i }));
+                        if (data[key] !== undefined) {
+                            sumValues += data[key];
+                            totalValues++;
+                        }
+                    }
+
+                    if (totalValues >= 5) {
+                        threshold = sumValues / totalValues;
+                    }
+                }
+
+                if (threshold == undefined) { return; }
+
+                let fillPercent = data[dayKey] / (threshold * 2);
+                if (fillPercent > 1) { fillPercent = 1; }
+                computedResults.push({
+                    dataType: dataType,
+                    fillPercent: fillPercent,
+                    value: dataType.formatter(data[dayKey])
+                });
+                setResults(computedResults);
+            });
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, props.date, props.dataTypes], 0);
+
+    if (result == null || !result.length) {
+        return null;
+    }
+
+    function getColor(dataType: RelativeActivityDataType, fillPercent: number) {
+        if (props.specifyThresholds && dataType.overThresholdColor && fillPercent > .5) {
+            return dataType.overThresholdColor;
+        }
+        return dataType.color || "var(--mdhui-color-primary)";
+    }
+
+    return <div ref={props.innerRef} className="mdhui-relative-activity">
+        {props.title && <CardTitle title={props.title} />}
+        {result.map(c => <div key={c.dataType.dailyDataType} className="mdhui-relative-activity-datatype">
+            <ActivityMeter className="mdhui-relative-activity-meter"
+                label={c.dataType.label}
+                value={c.value}
+                fillPercent={c.fillPercent}
+                averageFillPercent={0.5}
+                icon={c.dataType.icon}
+                color={getColor(c.dataType, c.fillPercent)} />
+            {c.dataType.threshold !== undefined &&
```

Comment:
I think there exists a couple of edge cases we'll want to account for here (and maybe below).

The first is that this will display if `c.dataType.threshold` is specified, `props.specifyThresholds` is `false`, and there were at least 5 values for the data type.  In that case, the actual threshold in use will be the 30 day average threshold, not this specified one.

The second is similar, with `c.dataType.threshold` being specified, `props.specifyThresholds` set to `false`, and there were less than 5 values for the data type.  It will then be using this specified threshold as the fallback, making this label correct, yet we'll also have the label below saying the midline for all the activities are their 30 day averages.